odrive_interface/src/odrive_interface/odrive_motor.py

```diff
@@ -1,14 +1,26 @@
 
-
+from odrive import enums
 import rospy
 
 
-TYPE_ERROR_MSG = 'Given type {gt} for {nm} is insufficient, requested type is {rt}'
+class OdriveMotor(object):
+    """Base class to communicate with a specific motor connected to a specified Odrive."""
+
+    AXIS_STATES = {k: v for (k, v) in enums.__dict__.items() if k.startswith('AXIS')}
 
+    MOTOR_MODES = {k: v for (k, v) in enums.__dict__.items() if k.startswith('MOTOR')}
+    CONTROL_MODES = {k: v for (k, v) in enums.__dict__.items() if k.startswith('CTRL')}
+    ENCODER_MODES = {k: v for (k, v) in enums.__dict__.items() if k.startswith('AXIS')}
+
+    AXIS_ERRORS = {k: v for (k, v) in enums.errors.axis.__dict__.items() if k.startswith('ERROR')}
+    MOTOR_ERRORS = {k: v for (k, v) in enums.errors.motor.__dict__.items() if k.startswith('ERROR')}
+    ENCODER_ERRORS = {k: v for (k, v) in enums.errors.encoder.__dict__.items() if k.startswith('ERROR')}
+    CONTROL_ERRORS = {k: v for (k, v) in enums.errors.controller.__dict__.items() if k.startswith('ERROR')}
```

Comment:
Hmm zou het niet netter zijn om k hier "error" te noemen en daarboven "mode"?

Reply:
indeed it would, fixed.